test/index.test.js

```diff
@@ -920,6 +919,53 @@ describe('Google Analytics', function() {
           analytics.deepEqual(toArray(window.ga.args[4]), ['send', 'event', 'cat 1', 'product clicked', { nonInteraction: 1 }]);
         });
 
+        it('should send product list viewed data', function() {
+          analytics.track('Product List Viewed', {
+            category: 'cat 1',
+            list_id: '1234',
+            products: [
+            { product_id: '507f1f77bcf86cd799439011' }
+            ]
+          });
+          analytics.assert(window.ga.args.length === 5);
+          analytics.deepEqual(toArray(window.ga.args[1]), ['set', '&cu', 'USD']);
+          analytics.deepEqual(toArray(window.ga.args[2]), ['ec:addImpression', {
+            id: '507f1f77bcf86cd799439011',
+            category: 'cat 1',
+            list: '1234',
+            position: 0
+          }]);
+          analytics.deepEqual(toArray(window.ga.args[3]), ['send', 'pageview']);
+          analytics.deepEqual(toArray(window.ga.args[4]), ['send', 'event', 'cat 1', 'Product List Viewed', { nonInteraction: 1 }]);
+        });
+
+        it('should send product list filtered data', function() {
+          analytics.track('Product List Filtered', {
+            category: 'cat 1',
+            list_id: '1234',
+            filters: [    {
+              type: 'department',
+              value: 'beauty'
+            },
+            {
+              type: 'price',
+              value: 'under'
+            }],
+            sorts:[ {
+              type: 'price',
+              value: 'desc'
+            }],
+            products: [
+            { product_id: '507f1f77bcf86cd799439011' }
+            ]
+          });
+          analytics.assert(window.ga.args.length === 5);
+          analytics.deepEqual(toArray(window.ga.args[1]), ['set', '&cu', 'USD']);
+          analytics.deepEqual(toArray(window.ga.args[2]),['ec:addImpression', { id: '507f1f77bcf86cd799439011', category: 'cat 1', variant: 'department:beauty,price:underprice:desc', list: '1234', position: 0 }]);
```

Comment:
Assuming that this isn't in a spec that I've missed, adding an additional "::" between the filters and sorters should be helpful for parsing down the line! :)